### ISSUE TITLE
Rachford-Rice solver returns V=1 for wide-K-spread mixtures (unbracketed Newton)

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#169]
### Summary

`_solve_rachford_rice` in `src/difflow/eos.py` uses an **unbracketed** `optimistix.Newton` started at `V0 = 0.5` and clips the result to `[0, 1]`. For a mixture with a wide K-value spread the Rachford-Rice function is stiff (poles at `V = 1/(1 - K_i)` just outside `[0, 1]`). Newton overshoots a pole, diverges, and the clipped result is `1.0`. `flash_TP_eos` then reports a **spurious single-phase vapor** even when the true vapor fraction is ~0.89. This throws downstream enthalpy/entropy (and any cooler/valve/expander temperature) off by 7-11 K, and likely contributes to cryogenic-column convergence failures.

### Reproducer (cryogenic NGL feed, PR EOS)

```python
import jax.numpy as jnp
from difflow.eos import PengRobinson, flash_TP_eos
from difflow.database import get_critical_props
names = ["nitrogen","carbon_dioxide","methane","ethane","propane",
         "isobutane","n_butane","isopentane","n_pentane","n_hexane"]
eos = PengRobinson({c: get_critical_props(c) for c in names})
z = jnp.array([0.005,0.007,0.860,0.070,0.030,0.005,0.010,0.003,0.005,0.005]); z=z/z.sum()
V, x, y = flash_TP_eos(eos, z, 233.15, 50e5)
print(float(V))   # -> 1.0   (reference PR: 0.89, genuinely two-phase)
```

The converged K-values are physically correct (methane K~3, hexane K~0.003, `sum(z/K) = 3.8 > 1`, i.e. below the dew point) - only the RR root solve is wrong.

### Fix

Use a bracketed / safeguarded RR solver (bisection, or the Leibovici-Neoschil window `V in (1/(1-Kmax), 1/(1-Kmin))`), with bubble/dew sign guards for single-phase feeds. A working differentiable prototype (bisection under `stop_gradient` + one analytic Newton step for exact implicit gradients) is in the `ngl-difflow` port (`ngl_difflow/flash_ext.py:rachford_rice`). With this fix, difflow's own PR fugacity + successive substitution converge to VF = 0.893 at the condition above (reference `thermo`: 0.890).

Found while reproducing a DWSIM NGL demethanizer base case in difflow.

**Comment by jkitchin** [comment]
Fixed in 301ee41: replaced the unbracketed Newton Rachford-Rice solver with a bracketed bisection on [0,1] (every pole lies outside the interval) plus one analytic Newton step for exact implicit gradients. The cryogenic NGL reproducer now gives V=0.883 instead of a spurious 1.0. Regression test added.
